From 689374bec40ec33bf920f319880bca8c02f8e301 Mon Sep 17 00:00:00 2001
From: Tomer Maimon <tmaimon77@gmail.com>
Date: Wed, 11 Sep 2024 20:01:44 +0300
Subject: [PATCH 23/23] arm64: dts: nuvoton: npcm845: set FIU UMA read

Set default UMA read instead of direct read for FIU0, FIU1 and FIU3 in
NPCM8XX BMC common device tree by removing FIUs memory mapping.

This change was made because FIU direct read can cause overloads
in the internal bus.

Signed-off-by: Tomer Maimon <tmaimon77@gmail.com>
---
 .../boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi  | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi b/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
index ce450af5d13b..e25b49319091 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
@@ -194,9 +194,8 @@ fiu0: spi@fb000000 {
 			compatible = "nuvoton,npcm845-fiu";
 			#address-cells = <1>;
 			#size-cells = <0>;
-			reg = <0x0 0xfb000000 0x0 0x1000>,
-			<0x0 0x80000000 0x0 0x10000000>;
-			reg-names = "control", "memory";
+			reg = <0x0 0xfb000000 0x0 0x1000>;
+			reg-names = "control";
 			clocks = <&clk NPCM8XX_CLK_SPI0>;
 			clock-names = "clk_ahb";
 			status = "disabled";
@@ -206,9 +205,8 @@ fiu1: spi@fb002000 {
 			compatible = "nuvoton,npcm845-fiu";
 			#address-cells = <1>;
 			#size-cells = <0>;
-			reg = <0x0 0xfb002000 0x0 0x1000>,
-			<0x0 0x90000000 0x0 0x4000000>;
-			reg-names = "control", "memory";
+			reg = <0x0 0xfb002000 0x0 0x1000>;
+			reg-names = "control";
 			clocks = <&clk NPCM8XX_CLK_SPI1>;
 			clock-names = "clk_spi1";
 			pinctrl-names = "default";
@@ -220,9 +218,8 @@ fiu3: spi@c0000000 {
 			compatible = "nuvoton,npcm845-fiu";
 			#address-cells = <1>;
 			#size-cells = <0>;
-			reg = <0x0 0xc0000000 0x0 0x1000>,
-			<0x0 0xA0000000 0x0 0x20000000>;
-			reg-names = "control", "memory";
+			reg = <0x0 0xc0000000 0x0 0x1000>;
+			reg-names = "control";
 			clocks = <&clk NPCM8XX_CLK_SPI3>;
 			clock-names = "clk_spi3";
 			pinctrl-names = "default";
-- 
2.43.0

